### PR TITLE
Custodian page

### PR DESCRIPTION
**Pull request by justin-ys**
## Ticket(s)
DIG-1117

## Description
Adds a new page for ingest to the data portal.
TO BE COMPLETED:
- Adds a page to specify permissions for the OPA dataset
- Adds a clinical ingest page
- Adds a genomic ingest page
- Connects with the ingest microservice (https://github.com/CanDIG/candigv2-ingest)

## Expected Behaviour

Using the new ingest page, users should be able to create a new OPA dataset with a list of authorized users for read access, upload a single-cohort JSON file for validation and ingest to this dataset, and add genomic data as well.

## Types of Change(s)
- [ ] 🪲 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

## Has it been tested for:
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] Prettier linter doesn't return errors
- [ ] Production branch PR browser testing: Chrome, Firefox, Edge, etc. 
- [ ] Locally tested
- [ ] Dev server tested
- [ ] Production tested when merging into stable/production branch

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
                 "highcharts": "^10.0.0",
                 "highcharts-react-official": "^3.1.0",
                 "history": "^5.0.0",
+                "js-cookie": "^3.0.5",
```
Comment (OrdiNeu) [inline package-lock.json]:
```suggestion
```
Per Son's concerns that cookies are disabled in Safari by default, we're removing js-cookie (and cookies in general) in favor of local storage.

File: package-lock.json [hunk 2]
```diff
@@ -13465,6 +13466,14 @@
...
             }
         },
+        "node_modules/js-cookie": {
+            "version": "3.0.5",
+            "resolved": "https://registry.npmjs.org/js-cookie/-/js-cookie-3.0.5.tgz",
+            "integrity": "sha512-cEiJEAEoIbWfCZYKWhVwFuvPX1gETRYPw6LlaTKoxD3s2AkXzkCjnp6h0V77ozyqj0jakteJ4YqDJT830+lVGw==",
+            "engines": {
+                "node": ">=14"
+            }
+        },
```
Comment (OrdiNeu) [inline package-lock.json]:
```suggestion
```

File: package-lock.json [hunk 3]
```diff
@@ -32964,6 +32973,11 @@
                 }
             }
         },
+        "js-cookie": {
+            "version": "3.0.5",
+            "resolved": "https://registry.npmjs.org/js-cookie/-/js-cookie-3.0.5.tgz",
+            "integrity": "sha512-cEiJEAEoIbWfCZYKWhVwFuvPX1gETRYPw6LlaTKoxD3s2AkXzkCjnp6h0V77ozyqj0jakteJ4YqDJT830+lVGw=="
+        },
```
Comment (OrdiNeu) [inline package-lock.json]:
```suggestion
```

File: src/ui-component/ingest/IngestTab.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    const setTab = (event, val) => {
+        setValue(val);
+    };
+
+    function getPage(val) {
+        if (val === 1) return <ClinicalIngestPage />;
+        if (val === 2) return <ClinicalIngestPage />;
+        if (val === 3) return <ClinicalIngestPage />;
+        return <ClinicalIngestPage />;
+    }
```
Comment (OrdiNeu) [inline src/ui-component/ingest/IngestTab.js]: This always returns one value -- is one or more of these supposed to be different?

File: src/ui-component/ingest/IngestTab.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+                centered
+            >
+                {tabs.map((name, idx) => (
+                    <IngestTab
+                        label={name}
+                        key={name}
+                        sx={{
+                            border: idx === value ? '0.15vw #2196F3 solid' : '0.1vw #2196F3 solid',
+                            borderBottom: idx === value ? 'none' : '0.15vw #2196F3 solid',
+                            background: idx === value ? '#FFFFFF' : '#F4FAFF'
```
Comment (OrdiNeu) [inline src/ui-component/ingest/IngestTab.js]: Makes more sense here to have two styles classes and switch between them depending on `idx === value`.

File: src/views/ingest/ingest.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            wordWrap: 'break-word',
+            marginLeft: '0.65em',
+            marginBottom: '1em',
+            marginTop: '0.51em'
+        }
+    });
+    const classes = useStyles();
+    return (
+        <Grid container sx={{ height: '100%', width: '100%' }}>
+            <Grid item sx={{ background: 'white' }} className={classes.container}>
```
Comment (OrdiNeu) [inline src/views/ingest/ingest.js]: Better to put background: 'white' in the container class, here, unless there's something later on you need to reuse the class for (with a non-white background).

File: src/ui-component/ingest/IngestTab.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    wordWrap: 'break-word',
+    textTransform: 'none',
+    verticalAlign: 'center'
+});
+
+export default function IngestMenu() {
+    const [value, setValue] = useState(0);
+    const setTab = (event, val) => {
+        setValue(val);
+    };
```
Comment (OrdiNeu) [inline src/ui-component/ingest/IngestTab.js]: Looks unnecessary (could just have an arrow function in Tabs' onChange), unless you were planning on adding more when the tab changes.

File: src/ui-component/ingest/IngestTab.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    color: '#686969',
+    fontSize: '1.5em',
+    fontFamily: 'Roboto',
+    fontWeight: '700',
+    wordWrap: 'break-word',
+    textTransform: 'none',
+    verticalAlign: 'center'
+});
+
+export default function IngestMenu() {
```
Comment (OrdiNeu) [inline src/ui-component/ingest/IngestTab.js]: Your function is the whole IngestMenu, but the file name is just the IngestTab -- might want to either split this up into two files or edit the filename so it's clear (remember that `export default` means that they will be calling it via `import IngestTab from 'ui-component/ingest/IngestTab.js'`, so it looks like it's the tab and not the menu)

Comment (OrdiNeu) [reply src/ui-component/ingest/IngestTab.js]: Also it's more common to define the function and then export default afterwards, as we often use PropTypes to help inform the IDE what props are meant to be passed.

**Review by OrdiNeu — CHANGES_REQUESTED** [review]
Couple of code style changes below. Looks good otherwise though. :heavy_check_mark:

File: src/ui-component/ingest/IngestTab.js (hunk 4 above)
Comment (justin-ys) [reply src/ui-component/ingest/IngestTab.js]: This is just filler

**Comment by OrdiNeu** [comment]
NB: The old (pre-rebase) version is now stored at `custodian-page-backup`, in case my rebase broke anything

**Review by OrdiNeu — APPROVED** [review]

**Pull request merged**